pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -106,6 +110,11 @@ func (c *Cluster) Start() error {
 		return fmt.Errorf("%v", err)
 	}
 
+	// fail if we were instructed to deploy more than one mgr on the same machine with host networking
+	if c.HostNetwork && c.allowMultiplePerNode && c.Replicas > 1 {
+		return fmt.Errorf("refusing to deploy %d managers on the same host since hostNetwork is %v and allowMultiplePerNode is %v. Only one manager per node is allowed", c.Replicas, c.HostNetwork, c.allowMultiplePerNode)
+	}
+
```

Comment:
Could we add a taint to the mgr deployments so they won't deploy on the same node? There isn't anything right now that ensures they are scheduled independently IIRC.